lib/src/source/git.dart

```diff
@@ -40,6 +41,49 @@ class GitSource extends CachedSource {
     });
   }
 
+  /// Gets the list of all versions from git tags.
+  /// Anything that Version.parse understands is considered a version,
+  /// it will also attempt to strip off a preceding 'v' e.g. v1.0.0
+  Future<List<Pubspec>> getVersions(String name, description) async {
+    if (!_useVersionTags(description)) {
+      // No need to get versions if the useVersionTags option is false
+      return super.getVersions(name, description);
+    }
+
+    PackageId id =
+        new PackageId(name, this.name, null, _getDescription(description));
+    String cachePath = _repoCachePath(id);
+    if (!entryExists(cachePath)) {
+      // Must have the repo cloned in order to list its tags
+      await _clone(_getUrl(id), cachePath);
+    }
+
+    List results = await git.run(["tag", "-l"], workingDir: cachePath);
+    List<Pubspec> validVersions = [];
+    for (String version in results) {
+      // Strip preceding 'v' character so 'v1.0.0' can be parsed into a Version
+      if (version.startsWith('v')) {
+        version = version.substring(1);
+      }
+      try {
+        // Use Version.parse to determine valid version tags
+        Version validVersion = new Version.parse(version);
+        // Fetch the pubspec for this version
+        Pubspec pubspec = await _getPubspec(cachePath, validVersion.toString());
+        if (pubspec != null) {
+          validVersions.add(pubspec);
+        }
+      } on FormatException {}
```

Comment:
Consider the case where a repo initially was tagged 1.0.0 without a pubspec (perhaps it used to be in JS or something ¯\_(ツ)_/¯ ). But all tags after 2.0.0 have a pubspec.
If we don't handle invalid pubspec here, this will cause an exception attempting to load the pubspecs for versions <2.0.0 and they won't be able to use version resolution at all for that given repo.